core/src/main/java/bio/terra/pearl/core/dao/search/EnrolleeSearchExpressionDao.java

```diff
@@ -26,20 +26,34 @@
 import org.jooq.SQLDialect;
 import org.jooq.impl.DSL;
 import org.springframework.stereotype.Component;
+import org.springframework.util.StopWatch;
 
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
-import java.util.function.Consumer;
+import java.util.function.BiConsumer;
 
 @Component
+@Slf4j
 public class EnrolleeSearchExpressionDao {
     private final Jdbi jdbi;
     private final EnrolleeDao enrolleeDao;
     private final ProfileDao profileDao;
 
+    /** list of mappers for the various modules that can be included in a search result */
+    protected static final List<SearchModuleMapper<? extends BaseEntity>> moduleMappers = List.of(
+            new SearchModuleMapper<>("enrollee", Enrollee.class, EnrolleeSearchExpressionResult::setEnrollee),
+            new SearchModuleMapper<>("profile", Profile.class, EnrolleeSearchExpressionResult::setProfile),
+            new SearchModuleMapper<>("portalUser", PortalParticipantUser.class, EnrolleeSearchExpressionResult::setPortalParticipantUser),
+            new SearchModuleMapper<>("participant_user", ParticipantUser.class, EnrolleeSearchExpressionResult::setParticipantUser),
+            new SearchModuleMapper<>("mailing_address", MailingAddress.class, EnrolleeSearchExpressionResult::setMailingAddress),
+            new SearchModuleMapper<>("latest_kit", KitRequest.class, EnrolleeSearchExpressionResult::setLatestKit),
+            new SearchModuleCollectionMapper<>("answer", Answer.class, (result, answer) -> result.getAnswers().add(answer)),
+            new SearchModuleCollectionMapper<>("task", ParticipantTask.class, (result, task) -> result.getTasks().add(task)));
```

Comment:
love this!!!

Reply:
thanks!  I spent a little bit of time seeing if I could avoid having these listed here -- if the query parser returned to this service a list of terms, then we'd both eliminate this list and reduce some unneccesary result set processing